kubetest/e2e.go

```diff
@@ -392,8 +394,15 @@ func DumpClusterLogs(location string) error {
 }
 
 func DumpFederationLogs(location string) error {
-	log.Printf("Dumping Federation logs to: %v", location)
-	return finishRunning(exec.Command("./federation/cluster/log-dump.sh", location))
+	logDumpPath := "./federation/cluster/log-dump.sh"
+	// federation/cluster/log-dump.sh only exists in the Kubernetes tree
+	// post-1.6. If it doesn't exist, do nothing and do not report an error.
+	if i, err := os.Stat(logDumpPath); err == nil && !i.IsDir() {
```

Comment:
Checking IsDir seemed odd to me too, but I was copying what someone else did elsewhere in this repo: https://github.com/kubernetes/test-infra/blob/72232020680b657f2f37b18bd5f46fa42ece272b/kubetest/extract.go#L234

Not an excuse, but I assumed that there was a good reason for that. @fejta, do you know why?